flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/validation/DefaultValidator.java

```diff
@@ -430,16 +420,24 @@ private Optional<String> validateNotLastStateUpgradeMode(FlinkSessionJob session
     private Optional<String> validateSpecChange(FlinkSessionJob sessionJob) {
         FlinkSessionJobSpec newSpec = sessionJob.getSpec();
 
-        if (sessionJob.getStatus() == null
-                || sessionJob.getStatus().getReconciliationStatus() == null
-                || sessionJob.getStatus().getReconciliationStatus().getLastReconciledSpec()
-                        == null) {
+        if (sessionJob.getStatus().getReconciliationStatus().getLastReconciledSpec() == null) {
```

Comment:
This could be improved to `sessionJob.getStatus().getReconciliationStatus().isFirstDeployment()`

Reply:
Fixed